diff --git a/biome/_biomes_all.xml b/biome/_biomes_all.xml
index dc29e9e..c7e0ce0 100644
--- a/biome/_biomes_all.xml
+++ b/biome/_biomes_all.xml
@@ -877,6 +877,12 @@
 		color="ff1f3b62" >
 	</Biome>
 	
+	<Biome 
+		biome_filename="data/biome/ghost_secret.xml" 
+		height_index="13"
+		color="ff1f3b64" >
+	</Biome>
+	
 	<Biome 
 		biome_filename="data/biome/clouds.xml" 
 		height_index="0"
diff --git a/biome_impl/biome_map.png b/biome_impl/biome_map.png
index 70cffc2..7c2dd63 100644
Binary files a/biome_impl/biome_map.png and b/biome_impl/biome_map.png differ
diff --git a/entities/animals/ethereal_being.xml b/entities/animals/ethereal_being.xml
index 25cd618..046efc3 100644
--- a/entities/animals/ethereal_being.xml
+++ b/entities/animals/ethereal_being.xml
@@ -1,4 +1,4 @@
-<Entity tags="enemy,teleportable_NOT,polymorphable_NOT,hittable,mortal" name="$animal_ethereal_being">	
+<Entity tags="enemy,teleportable_NOT,polymorphable_NOT,hittable,mortal,boss_ghost_helper" name="$animal_ethereal_being">	
 	<ItemChestComponent level="4" enemy_drop="1" > </ItemChestComponent>
 
 	<!-- art - main body ---------------------- -->
diff --git a/entities/buildings/statue_hand_1.xml b/entities/buildings/statue_hand_1.xml
index 7f3ea05..2c31619 100644
--- a/entities/buildings/statue_hand_1.xml
+++ b/entities/buildings/statue_hand_1.xml
@@ -1,4 +1,4 @@
-<Entity tags="statue_hand_1,statue_hand">
+<Entity tags="statue_hand_1,statue_hand,hittable">
 	<PhysicsBody2Component 
 		angular_damping="0.5"
 		destroy_body_if_entity_destroyed="1"
@@ -24,6 +24,14 @@
 		execute_every_n_frame="16"
 		>
 	</LuaComponent>
+	
+	<HitboxComponent 
+		damage_multiplier="0.1"
+		aabb_max_x="12" 
+		aabb_max_y="12" 
+		aabb_min_x="-12" 
+		aabb_min_y="-12" >
+	</HitboxComponent>
 
 </Entity>
 
diff --git a/materials.xml b/materials.xml
index de4c23c..c9f6e7f 100644
--- a/materials.xml
+++ b/materials.xml
@@ -8014,6 +8014,48 @@
 	color="ffE8BD5A" >
     </Graphics>
   </CellData>
+
+  <CellData 
+	name="glue"
+	ui_name="$mat_glue"
+  	tags="[liquid],[burnable],[alchemy],[impure]"
+	burnable="1"  
+	density="6"
+	durability="4"
+	hp="100"
+	cell_type="liquid" 
+	wang_color="fff6fce3"  
+	generates_smoke="0"
+	liquid_gravity="0.2"  
+	liquid_sand="1"  
+	liquid_viscosity="0"
+	liquid_damping="0.9"
+	liquid_sticks_to_ceiling="100"
+	liquid_flow_speed="0.1"
+	audio_physics_material_event="slime"
+	audio_physics_material_wall="slime"
+	audio_physics_material_solid="slime"
+	stickyness="1.0"
+	solid_friction="1.0"
+	fire_hp="50"
+	on_fire="0"
+	temperature_of_fire="95"
+	autoignition_temperature="10"
+	electrical_conductivity="0"
+	requires_oxygen="1"  
+	temperature_of_fire="10"
+	show_in_creative_mode="1"
+	>
+	<StatusEffects>
+		<Ingestion>
+			<StatusEffect type="FOOD_POISONING" amount="0.2" />
+			<StatusEffect type="TRIP" amount="0.3" />
+		</Ingestion>
+	</StatusEffects>
+    <Graphics
+		color="fff6fce3" >
+    </Graphics>
+  </CellData>
   
   <CellData 
 	name="slime"
@@ -12357,6 +12399,20 @@
     </Graphics>
   </CellDataChild>
   
+  <CellDataChild
+  	_parent="gem_box2d"
+  	_inherit_reactions="1"
+  	name="gem_box2d_white"
+	ui_name="$mat_gem_box2d_white"
+	gfx_glow_color="0xFFebeaef"
+	wang_color="ffebeaef"
+	show_in_creative_mode="1"
+	>
+  	<Graphics
+	color="ffebeaef" >
+    </Graphics>
+  </CellDataChild>
+  
   <CellDataChild
   	_parent="gem_box2d"
   	_inherit_reactions="1"
@@ -14179,6 +14235,14 @@
 	output_cell1="alcohol_gas"	output_cell2="alcohol_gas">
   </Reaction>
 
+  <!-- glue. evaporates slowly and burns quick into poison gas -->
+  <Reaction probability="60"
+  	fast_reaction="1"
+	input_cell1="glue" 		input_cell2="[fire]"
+	output_cell1="[fire]"		output_cell2="poison_gas">
+  </Reaction>
+
+
 	<!-- =================================================================================================================  -->
 	<!-- [slime] reactions -->
 
diff --git a/scripts/biomes/mountain_tree.lua b/scripts/biomes/mountain_tree.lua
index 835aa25..47f851d 100644
--- a/scripts/biomes/mountain_tree.lua
+++ b/scripts/biomes/mountain_tree.lua
@@ -296,7 +296,7 @@ function spawn_pillars( x, y )
 		{ { "misc_chest_rain", "crain" }, { "misc_worm_rain", "wrain" }, { "misc_greed_rain", "grain" }, { "misc_altar_tablet", "train" }, { "misc_monk_bots", "mbots" }, { "secret_tower", "secrett" }, { "player_status_ghostly", "pghost" }, { "player_status_ratty", "prat" } },
 		{ { "essence_fire", "essencef" }, { "essence_water", "essencew" }, { "essence_laser", "essencee" }, { "essence_air", "essencea" }, { "essence_alcohol", "essenceal" }, { "secret_moon", "moon" }, { "secret_moon2", "moona" }, { "special_mood", "moong" }, { "secret_dmoon", "dmoon" }, { "dead_mood", "dmoong" } },
 		{ { "progress_ending0", "end0" }, { "progress_ending1_toxic", "endt" }, { "progress_ending1_gold", "endb" }, { "progress_ending2", "endg" }, { "progress_newgameplusplus3", "endp" }, { "progress_nightmare", "endn" } },
-		{ { "miniboss_dragon", "minid" }, { "miniboss_limbs", "minil" }, { "miniboss_pit", "minip" }, { "miniboss_alchemist", "minia" }, { "miniboss_wizard", "meme" }, { "miniboss_gate_monsters", "minigm" }, { "boss_centipede", "boss" } },
+		{ { "miniboss_dragon", "minid" }, { "miniboss_limbs", "minil" }, { "miniboss_ghost", "minigh" }, { "miniboss_pit", "minip" }, { "miniboss_alchemist", "minia" }, { "miniboss_wizard", "meme" }, { "miniboss_gate_monsters", "minigm" }, { "boss_centipede", "boss" } },
 		{ { "progress_orb_1", "orbf" }, { "progress_orb_evil", "orbe" }, { "progress_orb_all", "orba" }, { "progress_pacifist", "pacifist" }, { "progress_nogold", "nogold" }, { "progress_clock", "clock" }, { "progress_minit", "minit" }, { "progress_nohit", "nohit" } },
 		{ { "secret_greed", "secretg" }, { "final_secret_orb", "yeah" }, { "secret_chest_dark", "secretcd" }, { "secret_chest_light", "secretcl" }, { "card_unlocked_everything", "secretall" }, { "card_unlocked_divide", "secretten" }, { "secret_fruit", "secretf" }, { "secret_allessences", "secretae" }, { "secret_meditation", "secretme" }, { "secret_buried_eye", "secretbe" }, { "secret_hourglass", "secrethg" }, { "progress_hut_a", "huta" }, { "progress_hut_b", "hutb" } },
 	}
diff --git a/scripts/biomes/snowcave.lua b/scripts/biomes/snowcave.lua
index cec8dd4..482f14b 100644
--- a/scripts/biomes/snowcave.lua
+++ b/scripts/biomes/snowcave.lua
@@ -297,6 +297,12 @@ g_big_enemies =
 		max_count	= 1,    
 		entity 	= "data/entities/animals/thunderskull.xml"
 	},
+	{
+		prob   		= 0.01,
+		min_count	= 2,
+		max_count	= 4,    
+		entity 	= "data/entities/animals/scavenger_glue.xml",
+	},
 	{
 		prob   		= 0.05,
 		min_count	= 1,
diff --git a/scripts/biomes/vault.lua b/scripts/biomes/vault.lua
index 0c5c623..6f55e9f 100644
--- a/scripts/biomes/vault.lua
+++ b/scripts/biomes/vault.lua
@@ -172,6 +172,12 @@ g_small_enemies =
 			"data/entities/animals/vault/coward.xml",
 		}
 	},
+	{
+		prob   		= 0.05,
+		min_count	= 1,
+		max_count	= 3,    
+		entity 	= "data/entities/animals/scavenger_glue.xml",
+	},
 }
 
 g_big_enemies =
@@ -257,6 +263,11 @@ g_big_enemies =
 				max_count	= 3,    
 				entity 	= "data/entities/animals/vault/scavenger_grenade.xml"
 			},
+			{
+				min_count	= 0,
+				max_count	= 3,    
+				entity 	= "data/entities/animals/vault/scavenger_glue.xml"
+			},
 			"data/entities/animals/vault/scavenger_leader.xml",
 			"data/entities/animals/vault/scavenger_heal.xml",
 		}
diff --git a/scripts/biomes/vault_frozen.lua b/scripts/biomes/vault_frozen.lua
index 141d996..fba4988 100644
--- a/scripts/biomes/vault_frozen.lua
+++ b/scripts/biomes/vault_frozen.lua
@@ -134,6 +134,7 @@ g_small_enemies =
 			"data/entities/animals/vault/scavenger_leader.xml",
 			"data/entities/animals/vault/scavenger_smg.xml",
 			"data/entities/animals/vault/scavenger_grenade.xml",
+			"data/entities/animals/vault/scavenger_glue.xml",
 			"data/entities/animals/vault/coward.xml",
 		},
 	},
diff --git a/scripts/gun/gun_actions.lua b/scripts/gun/gun_actions.lua
index d7fcd74..72a5a92 100644
--- a/scripts/gun/gun_actions.lua
+++ b/scripts/gun/gun_actions.lua
@@ -4488,7 +4488,7 @@ actions =
 		sprite_unidentified = "data/ui_gfx/gun_actions/bounce_unidentified.png",
 		type 		= ACTION_TYPE_MODIFIER,
 		spawn_level                       = "2,3,4,5,6", -- BOUNCE
-		spawn_probability                 = "1,1,1,1,1", -- BOUNCE
+		spawn_probability                 = "1,1,0.4,0.2,0.2", -- BOUNCE
 		price = 50,
 		mana = 0,
 		--max_uses = 150,
@@ -4497,6 +4497,24 @@ actions =
 			draw_actions( 1, true )
 		end,
 	},
+	{
+		id          = "REMOVE_BOUNCE",
+		name 		= "$action_remove_bounce",
+		description = "$actiondesc_remove_bounce",
+		sprite 		= "data/ui_gfx/gun_actions/remove_bounce.png",
+		sprite_unidentified = "data/ui_gfx/gun_actions/bounce_unidentified.png",
+		type 		= ACTION_TYPE_MODIFIER,
+		spawn_level                       = "2,3,4,5,6", -- BOUNCE
+		spawn_probability                 = "0.2,0.2,1,1,1", -- BOUNCE
+		price = 50,
+		mana = 0,
+		--max_uses = 150,
+		action 		= function()
+			c.extra_entities = c.extra_entities .. "data/entities/misc/remove_bounce.xml,"
+			c.bounces = 0
+			draw_actions( 1, true )
+		end,
+	},
 	{
 		id          = "HOMING",
 		name 		= "$action_homing",
diff --git a/scripts/projectiles/teleport_projectile_closer_hit.lua b/scripts/projectiles/teleport_projectile_closer_hit.lua
index 6c196e3..9230cb0 100644
--- a/scripts/projectiles/teleport_projectile_closer_hit.lua
+++ b/scripts/projectiles/teleport_projectile_closer_hit.lua
@@ -23,7 +23,7 @@ if ( #projectiles > 0 ) then
 		end
 	end
 	
-	print( tostring(entity_id) .. ", " .. tostring(enemy_id) )
+	-- print( tostring(entity_id) .. ", " .. tostring(enemy_id) )
 	
 	if ( ox ~= nil ) and ( oy ~= nil ) then
 		EntitySetTransform( enemy_id, ox, oy )
